deps/rabbitmq_amqp1_0/test/amqp10_client_SUITE.erl

```diff
@@ -151,6 +153,68 @@ reliable_send_receive(Config, Outcome) ->
 
     ok.
 
+publishing_to_non_existing_queue_should_settle_with_released(Config) ->
+    Container = atom_to_binary(?FUNCTION_NAME, utf8),
+    Suffix = <<"foo">>,
+    %% does not exist
+    QName = <<Container/binary, Suffix/binary>>,
+    Host = ?config(rmq_hostname, Config),
+    Port = rabbit_ct_broker_helpers:get_node_config(Config, 0, tcp_port_amqp),
+    Address = <<"/exchange/amq.direct/", QName/binary>>,
+
+    OpnConf = #{address => Host,
+                port => Port,
+                container_id => Container,
+                sasl => {plain, <<"guest">>, <<"guest">>}},
+    {ok, Connection} = amqp10_client:open_connection(OpnConf),
+    {ok, Session} = amqp10_client:begin_session(Connection),
+    SenderLinkName = <<"test-sender">>,
+    {ok, Sender} = amqp10_client:attach_sender_link(Session,
+                                                    SenderLinkName,
+                                                    Address),
+    ok = wait_for_credit(Sender),
+    DTag1 = <<"dtag-1">>,
+    %% create an unsettled message,
+    %% link will be in "mixed" mode by default
+    Msg1 = amqp10_msg:new(DTag1, <<"body-1">>, false),
+    ok = amqp10_client:send_msg(Sender, Msg1),
+    ok = wait_for_settlement(DTag1, released),
+
+    ok = amqp10_client:detach_link(Sender),
+    ok = amqp10_client:close_connection(Connection),
+    flush("post sender close"),
+    ok.
+
+open_link_to_non_existing_destination_should_end_session(Config) ->
+    Container = atom_to_list(?FUNCTION_NAME),
+    Name = Container ++ "foo",
+    Addresses = [
+                 "/exchange/" ++ Name ++ "/bar",
+                 "/amq/queue/" ++ Name
```

Comment:
@acogoluegnes should we also test target addresses `"/queue/" ++ Name` and `Name`?

Reply:
The queue is automatically created for these 2 kinds of addresses IIRC. Feel free to give it a try or have a look at the code, but I'm pretty that's the case.

Reply:
You're right, thanks.
I got confused by the test name `open_link_to_non_existing_destination_should_end_session`:
Opening a link to a non existing target should actually work for addresses `"/queue/" ++ Name` and `Name` and only fail non existing exchanges and `"/amq/queue/"`  because the latter has a special meaning:
> Queue names starting with "amq." are reserved for pre-declared and standardised queues.

Reply:
Sure, no problem. I kept the original behavior, a module enforces the existence of `/queue/` and `queue-name` addresses.